### PR TITLE
Fix mach hardware exception forwarding

### LINKED ISSUE #104968: exception_handling/pal_sxs/test1/paltest_pal_sxs_test1 failure in CI
Example pipeline run: https://dev.azure.com/dnceng-public/public/_build/results?buildId=742648&view=results
Example console log: https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-81a5f2c525bf47c9ac/PALTests/1/console.a86a01b0.log?helixlogtype=result

```scala
Running tests...

......................................................................................Testing for 64 Bit Platforms 
...................
FAILED: exception_handling/pal_sxs/test1/paltest_pal_sxs_test1. Exit code: 137
```

### PR DESCRIPTION
**Pull request by janvorli**
A recent change to enable AVX512 register state processing in the mach exception handling on macOS x64 has broken a PAL test and also a case when a hardware exception occurs in 3rd party native code. In both cases runtime hangs, another exception occurs while forwarding the exception message and that ends up happenning infinitely.

The problem is caused by the fact that the MachMessageInfo has grown, since we've changed a field containing float state to contain avx512 state instead. But the buffer that stores the message is of fixed size of 1500 bytes and it is not sufficient.

I have grown the buffer size to the necessary size, but there was another issue lurking behind the first one. The MachExceptionInfo::RestoreState was trying to restore the float state always as x86_FLOAT_STATE instead of choosing x86_FLOAT_STATE, x86_AVX_STATE or x86_AVX512_STATE depending of which of them was stored. So the thread_set_state was failing.

This change fixes both of the problems.

Close #104968

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

**Comment by janvorli** [comment]
@jkotas I've simplified the state restoring. I've found that the count is stored in the state as well, so I don't need to have the switch.

**Review by jkotas — APPROVED** [review]

File: src/coreclr/pal/src/exception/machmessage.h [hunk 1]
```diff
@@ -189,7 +189,7 @@ class MachMessage
     // The maximum size in bytes of any Mach message we can send or receive. Calculating an exact size for
     // this is non trivial (basically because of the security trailers that Mach appends) but the current
     // value has proven to be more than enough so far.
-    static const size_t kcbMaxMessageSize = 1500;
+    static const size_t kcbMaxMessageSize = 0x1500;
```
Comment (jkotas) [inline src/coreclr/pal/src/exception/machmessage.h:192]: Would it be better to define this as `sizeof(mach_message_t)` plus some extra space?

Comment (janvorli) [reply src/coreclr/pal/src/exception/machmessage.h:192]: I've actually found a constant in the headers representing the maximum trailer size, so we can define it as sizeof(mach_message_t) + MAX_TRAILER_SIZE and be safe.

Comment (janvorli) [reply src/coreclr/pal/src/exception/machmessage.h]: Added a commit with that change.

Comment (mikelle-rogers) [reply src/coreclr/pal/src/exception/machmessage.h]: Is the MAX_TRAILER_SIZE in a mach source file header? And I am guessing it refers to the security trailers that are mentioned in the comment above?

**Review by mikelle-rogers — APPROVED** [review]
Thanks for catching this!

File: src/coreclr/pal/src/exception/machmessage.h (hunk 1 above)
Comment (janvorli) [reply src/coreclr/pal/src/exception/machmessage.h]: Yes, it is in the Apple headers and it refers to those trailers.

**Pull request merged**